basis/basis.py

```diff
@@ -0,0 +1,172 @@
+import abc
+import numpy
+import math
```

Comment:
Per [PEP-8](https://peps.python.org/pep-0008/#imports), imports should come in the order: standard library, third-party, local with a blank line between groups.
```suggestion
import math

import numpy
```